test/e2e/registration_service_test.go

```diff
@@ -51,3 +56,415 @@ func (s *registrationServiceTestSuite) TestLandingPageReachable() {
+		err = json.Unmarshal([]byte(body), &mp)
+		require.Nil(s.T(), err)
+
+		// Verify JSON response.
+		alive := mp["alive"]
+		require.True(s.T(), alive.(bool))
+
+		environment := mp["environment"]
+		require.Equal(s.T(), "e2e-tests", environment.(string))
+
+		revision := mp["revision"]
+		require.NotNil(s.T(), revision)
+
+		buildTime := mp["buildTime"]
+		require.NotNil(s.T(), buildTime)
+
+		startTime := mp["startTime"]
+		require.NotNil(s.T(), startTime)
+	})
+}
+
+func (s *registrationServiceTestSuite) TestAuthConfig() {
+	s.Run("verify_authconfig", func() {
+		// Call authconfig endpoint.
+		req, err := http.NewRequest("GET", s.route+"/api/v1/authconfig", nil)
+		require.NoError(s.T(), err)
+		client := &http.Client{
+			Timeout: time.Second * 10,
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{
+					InsecureSkipVerify: true,
+				},
+			},
+		}
+
+		resp, err := client.Do(req)
+		require.NoError(s.T(), err)
+		assert.Equal(s.T(), http.StatusOK, resp.StatusCode)
+
+		defer resp.Body.Close()
+
+		body, err := ioutil.ReadAll(resp.Body)
+		require.Nil(s.T(), err)
+		require.NotNil(s.T(), body)
+
+		mp := make(map[string]interface{})
+		err = json.Unmarshal([]byte(body), &mp)
+		require.Nil(s.T(), err)
+
+		// Verify JSON response.
+		alive := mp["auth-client-library-url"]
+		require.Equal(s.T(), alive.(string), "https://sso.prod-preview.openshift.io/auth/js/keycloak.js")
+
+		testingMode := mp["auth-client-config"].(string)
+		mp1 := make(map[string]interface{})
+		err = json.Unmarshal([]byte(testingMode), &mp1)
+		require.Nil(s.T(), err)
+
+		realm := mp1["realm"]
+		require.Equal(s.T(), realm.(string), "toolchain-public")
+
+		authServerURL := mp1["auth-server-url"]
+		require.Equal(s.T(), authServerURL.(string), "https://sso.prod-preview.openshift.io/auth")
+
+		sslRequired := mp1["ssl-required"]
+		require.Equal(s.T(), sslRequired.(string), "none")
+
+		resource := mp1["resource"]
+		require.Equal(s.T(), resource.(string), "crt")
+
+		publicClient := mp1["public-client"]
+		require.True(s.T(), publicClient.(bool))
+	})
+}
+
+func (s *registrationServiceTestSuite) TestSignup() {
+	// Get valid generated token for e2e tests. IAT claim is overriden
+	// to avoid token used before issued error.
+	identity0 := authsupport.NewIdentity()
+	emailClaim0 := authsupport.WithEmailClaim(uuid.NewV4().String() + "@email.tld")
+	iatClaim0 := authsupport.WithIATClaim(time.Now().Add(-60 * time.Second))
+	token0, err := authsupport.GenerateSignedE2ETestToken(*identity0, emailClaim0, iatClaim0)
+	require.NoError(s.T(), err)
+
+	s.Run("verify_post_signup_error_no_token", func() {
+		// Call signup endpoint without a token.
+		requestBody, err := json.Marshal(map[string]string{})
+		require.NoError(s.T(), err)
+		req, err := http.NewRequest("POST", s.route+"/api/v1/signup", bytes.NewBuffer(requestBody))
+		require.NoError(s.T(), err)
+		req.Header.Set("content-type", "application/json")
+		client := &http.Client{
```

Comment:
you may want to create a func to init this client, since this code is duplicated quite a few times

Reply:
fixed here: 563467bf4c0dc459366a02e04e264965d15c61ca